src/sql/src/statements/transform/expand_interval.rs

```diff
@@ -91,7 +91,30 @@ impl TransformRule for ExpandIntervalTransformRule {
                     _ => {}
                 },
                 _ => {}
+            },
+            Expr::Cast {
+                expr: cast_exp,
+                data_type,
+                ..
+            } => {
+                if DataType::Interval == *data_type {
+                    match *cast_exp.clone() {
+                        Expr::Value(Value::SingleQuotedString(value))
+                        | Expr::Value(Value::DoubleQuotedString(value)) => {
+                            let data = expand_interval_name(&value).unwrap_or(value);
+                            *expr = Expr::Interval(Interval {
+                                value: single_quoted_string_expr(data),
+                                leading_field: None,
+                                leading_precision: None,
+                                last_field: None,
+                                fractional_seconds_precision: None,
+                            })
+                        }
+                        _ => {}
+                    }
+                }
             }
```

Comment:
The same suggestion as above, try to avoid clone:

```suggestion
Expr::Cast {
                expr: cast_exp,
                data_type,
                ..
            } => {
                if DataType::Interval == *data_type {
                    match &**cast_exp {
                        Expr::Value(Value::SingleQuotedString(value))
                        | Expr::Value(Value::DoubleQuotedString(value)) => {
                            let data =
                                expand_interval_name(value).unwrap_or_else(|| value.to_string());
                            *expr = Expr::Interval(Interval {
                                value: single_quoted_string_expr(data),
                                leading_field: None,
                                leading_precision: None,
                                last_field: None,
                                fractional_seconds_precision: None,
                            })
                        }
                        _ => {}
                    }
                }
            }
```